File: python/contador.py
'''
 * Crea un programa se encargue de transformar un número
 * decimal a binario sin utilizar funciones propias del lenguaje que lo hagan directamente.
 */
'''

# function to parse decimal to binary
def decimal_to_binary(decimal):
    number = decimal
    binary = ''

    while number != 0:
        reminder = number % 2
        number = number // 2

        binary = str(reminder) + binary

    return binary or '0'

# main function
def main():
    print(decimal_to_binary(319))
    print(decimal_to_binary(0))
    print(decimal_to_binary(3))

if __name__ == '__main__':
    main()

File: javascript/contador.js
/** 
 * Crea un programa se encargue de transformar un número
 * decimal a binario sin utilizar funciones propias del lenguaje que lo hagan directamente.
 */


/**
 * Function that returns the number on binary previously inputted
 * @param {integer} decimal 
 * @returns binary 
 */
function numberTransform(decimal) {
    let number  = decimal;
    let binary = '';

    while (number !== 0) {
        let reminder = number % 2;
        number = Math.floor(number / 2);
        binary = reminder.toString() + binary;
       
    }

    return binary || '0';
}

console.log(numberTransform(387));
console.log(numberTransform(0));